numerical_methods/jacobi_method.cpp

```diff
@@ -0,0 +1,170 @@
+/**
+ * @file
+ * @brief Implementation of [The Jacobi Method]
+ (https://en.wikipedia.org/wiki/Jacobi_method#Algorithm)
+ (https://college.cengage.com/mathematics/larson/elementary_linear/5e/students/ch08-10/chap_10_2.pdf)
+ * @details
+ * ### Algorithm
+ *
+ * This is an iterative method to solve the system of linear equations
+ * with two assumptions :-
+ * 1) system has a unique solution.
+ * 2) the coefficient matrix has not zeros on main diagonal.
+ *
+ * Implementation:
+ * model problem:
+ * a11*x + a12*y + a13*z = b1 |\
+ * a21*x + a22*y + a23*z = b2    >> Given system of Equation.
+ * a31*x + a32*y + a33*z = b3 |/
+ *
+ * Taking the Coefficient matrix A and Column vector B are as
+ *     |a11 a12 a13 |      |b1|
+ * A = |a21 a22 a23 | b =  |b2|
+ *     |a31 a32 a33 |      |b3|
+ *
+ * Then we have an initial approximation values variables in vector X =
+ * [x0,y0,z0] Iterations : (Note: in xi,yi,zi , i represent the iteration no.)
+ * x(i+1) = (b1 - a12*yi - a13*zi)/a11
+ * y(i+1) = (b2 - a21*xi - a23*zi)/a22
+ * z(i+1) = (b3 - a32*xi - a32*yi)/a33
+ *
+ * Please refer to the following link for more generic description of iterations
+ formulas
+ * https://mathworld.wolfram.com/JacobiMethod.html
+ *
+ * @author [Nitish Pal](https://github.com/NitishPal2013)
+ */
+
+#include <cassert>   /// for assert
+#include <cmath>     /// for pow() and round()
+#include <iostream>  /// for IO operations
+#include <vector>    /// for vector class to storing values
+
+using std::cout;
+using std::vector;
+
+/**
+ * @namespace jacobi
+ * @brief
+ */
+
+namespace jacobi {
+
+/**
+ *@brief jacobi_method documentation
+ * jacobi_method class is to implement [The Jacobi Method].
```

Comment:
Yes, we know it's used to implement the Jacobi Method, but what is the use of its functions/variables (should be a summary)? Why are those created and used?

Reply:
Okay Sir.